From a5079e4abc5afecc173d5a170b0bd71fc9133681 Mon Sep 17 00:00:00 2001
From: Jason Liu <r64343@freescale.com>
Date: Fri, 6 Sep 2013 12:37:02 +0800
Subject: [PATCH 420/435] ENGR00278625-1 fix the dump-clocks issues due to sys
 interface change

the sys for the clk interface has been changed, this patch update it
to reflect the new interface.

Signed-off-by: Jason Liu <r64343@freescale.com>
---
 test/mxc_clock_test/dump-clocks-dot.sh | 30 +++++++++++++++++++++---------
 test/mxc_clock_test/dump-clocks.sh     | 27 ++++++++++++++++++++-------
 2 files changed, 41 insertions(+), 16 deletions(-)

diff --git a/test/mxc_clock_test/dump-clocks-dot.sh b/test/mxc_clock_test/dump-clocks-dot.sh
index adba4b9..14afa36 100755
--- a/test/mxc_clock_test/dump-clocks-dot.sh
+++ b/test/mxc_clock_test/dump-clocks-dot.sh
@@ -13,34 +13,47 @@ fi
 
 saved_path=$PWD
 
-#printf "%-24s %-24s %3s %9s\n" "clock" "parent" "use" "rate"
 printf "digraph clocktree {\n"
 printf "rankdir = LR;\n"
 printf "node [shape=record];\n"
 
-for foo in $(find /sys/kernel/debug/clock -type d); do
-    if [ "$foo" = '/sys/kernel/debug/clock' ]; then
+for foo in $(find /sys/kernel/debug/clk -type d); do
+    if [ "$foo" = '/sys/kernel/debug/clk' ]; then
         continue
     fi
 
     cd $foo
 
-    use="$(cat usecount)"
-    rate="$(cat rate)"
+    if [ -f clk_flags ]; then
+        flags="$(cat clk_flags)"
+    fi
+
+    if [ -f clk_enable_count ]; then
+        enable_count="$(cat clk_enable_count)"
+    fi
+
+    if [ -f clk_prepare_count ]; then
+        prepare_count="$(cat clk_enable_count)"
+    fi
+
+    if [ -f clk_rate ]; then
+        rate="$(cat clk_rate)"
+    fi
 
     clk="$(basename $foo)"
     cd ..
     parent="$(basename $PWD)"
 
-    if [ "$use" = 0 ]; then
+    if [ "$enable_count" = 0 ]; then
     	bgcolor="red"
     else
     	bgcolor="green"
     fi
 
-    printf "%s [label=\"%s|rate %d|use $use\" color=%s];\n" "$clk" "$clk" "$rate" $bgcolor
+    printf "%s [label=\"%s|rate %d|en_cnt %d|pr_cnt %d|flags %08x\" color=%s];\n" \
+	"$clk" "$clk" "$rate" "$enable_count" "$prepare_count" "$flags" $bgcolor
 
-    if [ "$parent" = 'clock' ]; then
+    if [ "$parent" = 'clk' ]; then
         parent="virtual_Root"
 # 	printf "%s d];\n"  "$clk" "$clk" "$rate"
     else
@@ -49,7 +62,6 @@ for foo in $(find /sys/kernel/debug/clock -type d); do
 
 #    printf "%s [label=%s_%d];\n" "$clk" "$rate"
 #    printf "%s -> %s ;\n" "$parent" "$clk"
-# "$use" "$rate"
 
     cd $saved_path
 done
diff --git a/test/mxc_clock_test/dump-clocks.sh b/test/mxc_clock_test/dump-clocks.sh
index a57e6d2..09c4705 100755
--- a/test/mxc_clock_test/dump-clocks.sh
+++ b/test/mxc_clock_test/dump-clocks.sh
@@ -6,27 +6,40 @@ fi
 
 saved_path=$PWD
 
-printf "%-24s %-24s %3s %9s\n" "clock" "parent" "use" "rate"
+printf "%-24s %-24s %8s %6s %6s %9s\n" "clock" "parent" "flags" "en_cnt" "pre_cnt" "rate"
 
-for foo in $(find /sys/kernel/debug/clock -type d); do
-    if [ "$foo" = '/sys/kernel/debug/clock' ]; then
+for foo in $(find /sys/kernel/debug/clk -type d); do
+    if [ "$foo" = '/sys/kernel/debug/clk' ]; then
         continue
     fi
 
     cd $foo
 
-    use="$(cat usecount)"
-    rate="$(cat rate)"
+    if [ -f clk_flags ]; then
+        flags="$(cat clk_flags)"
+    fi
+
+    if [ -f clk_enable_count ]; then
+        enable_count="$(cat clk_enable_count)"
+    fi
+
+    if [ -f clk_prepare_count ]; then
+        prepare_count="$(cat clk_enable_count)"
+    fi
+
+    if [ -f clk_rate ]; then
+        rate="$(cat clk_rate)"
+    fi
 
     clk="$(basename $foo)"
     cd ..
     parent="$(basename $PWD)"
 
-    if [ "$parent" = 'clock' ]; then
+    if [ "$parent" = 'clk' ]; then
         parent="   ---"
     fi
 
-    printf "%-24s %-24s %2d %10d\n" "$clk" "$parent" "$use" "$rate"
+    printf "%-24s %-24s %08x %6d %6d %10d\n" "$clk" "$parent" "$flags" "$enable_count" "$prepare_count" "$rate"
 
     cd $saved_path
 done
-- 
1.8.0

